### PR TITLE
refactor: replace asm! with llvm_asm! and fix some clippy issues

### PR DESCRIPTION
**Pull request by 0xPoe**
Replace asm! with llvm_asm!

And fix some clippy issues, but still have some issues for now.  I will fix it after this PR and add fmt and clippy CI check.

### CODE REVIEW

**Comment by 0xPoe** [comment]
Pass all tests. @wangrunji0408  Could you please take a look? Thanks!

After this PR I will send a PR to modify the docs to use llvm_asm!.

File: rust-toolchain [hunk 1]
```diff
@@ -1 +1 @@
-nightly-2020-02-20
+nightly
```
Comment (jiegec) [inline rust-toolchain]: Please lock nightly to a newer date.

Comment (wangrunji0408) [reply rust-toolchain]: +1

Comment (0xPoe) [reply rust-toolchain]: Added.

**Pull request merged**